app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayDialog.kt

```diff
@@ -6,19 +6,18 @@ package org.mozilla.fenix.tabstray
 
 import android.app.Dialog
 import android.content.Context
-import org.mozilla.fenix.tabstray.browser.BrowserTrayInteractor
 
 /**
  * Default tabs tray dialog implementation for overriding the default on back pressed.
  */
 class TabsTrayDialog(
     context: Context,
     theme: Int,
-    private val interactor: () -> BrowserTrayInteractor,
+    private val interactor: () -> TabsTrayInteractor,
```

Comment:
I'd have to double check, but this smelled like a hack to get around a race condition or dependency issue, so that a lambda could be called after onStart, or something, to provide the interactor, rather than have the interactor as a compile time dependency